# Exception_handling

---
---

* [try_catch_1](#try_catch_1)



---
---

예외 처리 구문 ``try ~ catch``

---

* 에러(error) : 수습될 수 없는 심각한 오류

* 예외(exception) : 수습될 수 있는 다소 미약한 오류

---

* 컴파일 에러 : 컴파일 시에 발생하는 에러

* 런타임 에러 : 실행 시에 발생하는 에러

* 논리적 에러 : 실행은 되지만, 의도와 다르게 동작하는 것

---

* ``Exception 클래스들`` - 사용자의 실수와 같은 외적인 요인에 의해 발생하는 예외

* ``RuntimeException`` - 프로그래머의 실수에 의해 발생하는 예외

---

* 예외처리(Exception handling)의 정의
	
	* 프로그램 실행 시 발생할 수 있는 예외에 대비한 코드를 작성하는 것
	
* 예외처리(Exception handling)의 목적

	* 프로그램의 비정상 종료를 막고, 정상적인 실행상태를 유지하는 것
	
---
---

### try_catch_1

예외처리가 되지 않은 상황이 발생하면, 프로그램은 강제 종료된다.

예외상황이 발생하더라도 프로그램을 정상 동작하도록 처리를 할 수 있는데, 이러한 방법을 **예외처리**라고 한다.

``try ~ catch``구문을 사용하는데,

``try``에는 예외가 발생할 수 있는 문장을 넣고, ``try``안쪽에서 예외가 발생하게 되면,

``catch``문에서 해당 예외를 찾는다.

``catch``에서 예외가 처리될 경우에는 프로그램은 강제종료 되지 않고, 정상 동작을 유지하게 된다.

:camel:[Top](#exception_handling)